src/SingleArticlePage.jsx

```diff
@@ -2,22 +2,47 @@ import { useParams } from "react-router-dom";
 import { useState, useEffect } from 'react';
 import getSingleArticle from "./axiosURLs/getSingleArticle";
 import { CommentCards } from "./CommentCards";
+import axios from "axios";
+
 
 export function SingleArticlePage() {
 
   const { article_id } = useParams();
   const [singleArticleData, setSingleArticleData] = useState({});
   const [loading, setLoading] = useState(true);
+  const [voteCount, setVoteCount] = useState(0)
+  const [error, setError] = useState({})
+  
 
 
   useEffect(() => {
     getSingleArticle(article_id).then((response) => {
 
       setSingleArticleData(response.data.article[0]);
       setLoading(false);
+      setVoteCount(response.data.article[0].votes)    //I've set the initail votes value
     });
   }, []);
 
+  const patchVotes = (incVotes) => {
+    return axios.patch(`https://andrew-nc-news.onrender.com/api/articles/${article_id}`, { inc_votes: incVotes })
+      .then((response) => {
+        console.log(response.data.article.votes)
+        setVoteCount(response.data.article.votes)
+      }).catch((err) => {
+        setError(err)
+      })
+    }
+
+    const handleUpvote = () => {
+      patchVotes(1);
+    };
+  
+    const handleDownvote = () => {
+      patchVotes(-1);
+    };
```

Comment:
Ideally we'd use [optimistic rendering](https://notes.northcoders.com/courses/js-front-end/ui-behaviours-and-optimistic-rendering) in order to assume that the vote will be successful, change the vote total immediately (before the request succeeds), only reverting the vote if something goes wrong, check out the linked note page for a hand with this!